test/jasmine/tests/legend_test.js

```diff
@@ -1667,6 +1666,68 @@ describe('legend interaction', function() {
                 .then(done);
             });
         });
+
+        describe('should honor *itemclick* and *itemdoubleclick* settings', function() {
+            var _assert;
+
+            function run() {
+                return Promise.resolve()
+                    .then(click(0, 1)).then(_assert(['legendonly', true]))
+                    .then(click(0, 1)).then(_assert([true, true]))
+                    .then(click(0, 2)).then(_assert([true, 'legendonly']))
+                    .then(click(0, 2)).then(_assert([true, true]))
+                    .then(function() {
+                        return Plotly.relayout(gd, {
+                            'legend.itemclick': false,
+                            'legend.itemdoubleclick': false
+                        });
+                    })
+                    .then(click(0, 1)).then(_assert([true, true]))
+                    .then(click(0, 2)).then(_assert([true, true]))
+                    .then(function() {
+                        return Plotly.relayout(gd, {
+                            'legend.itemclick': 'focus',
+                            'legend.itemdoubleclick': 'toggle'
+                        });
+                    })
+                    .then(click(0, 1)).then(_assert([true, 'legendonly']))
+                    .then(click(0, 1)).then(_assert([true, true]))
+                    .then(click(0, 2)).then(_assert(['legendonly', true]))
+                    .then(click(0, 2)).then(_assert([true, true]));
+            }
+
+            it('- regular trace case', function(done) {
+                _assert = assertVisible;
+
+                Plotly.plot(gd, [
+                    { y: [1, 2, 1] },
+                    { y: [2, 1, 2] }
```

Comment:
Non-blocking but adding a third trace could be nice to clearly distinguish the behavior of `focus` versus `toggle`.
In `run()` we would have:
```javascript
.then(click(0, 1)).then(_assert(['legendonly', true, true]))
.then(click(0, 1)).then(_assert([true, true, true]))
.then(click(0, 2)).then(_assert([true, 'legendonly', 'legendonly']))
.then(click(0, 2)).then(_assert([true, true, true]))
```

Reply:
Good call -> https://github.com/plotly/plotly.js/pull/3862/commits/81907381b6615488751e4e01131d1899e04948c2